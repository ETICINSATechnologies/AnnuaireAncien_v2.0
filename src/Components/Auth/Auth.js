class Auth {
    static connect(jwt){
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(){
        sessionStorage.removeItem('token')
    }

    static isConnected(){
        return !!sessionStorage.getItem('token');
    }

    static getToken() {
        return sessionStorage.getItem('token')
    }

    static getPayload(){
        return sessionStorage.getItem('token');
    }

    static addCorrectButton(buttons){
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
