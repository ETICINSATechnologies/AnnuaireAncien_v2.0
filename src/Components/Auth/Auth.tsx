class Auth {
    static connect(jwt: string) {
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(): void {
        sessionStorage.removeItem('token')
    }

    static isConnected(): boolean {
        return !!sessionStorage.getItem('token');
    }

    static getToken(): string {
        return sessionStorage.getItem('token') as string;
    }

    static addCorrectButton(buttons: Array<string>) {
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
