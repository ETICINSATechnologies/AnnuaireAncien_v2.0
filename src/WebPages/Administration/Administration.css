.Administration {
    display: grid;
    height: 100%;
}

@media (orientation: landscape) {
    .Administration {
        grid-template-columns: 1fr 1fr 1fr;
    }

    .Administration caption {
        font-size: 1.5vw;
    }

    .Administration tbody {
        max-height: calc(85vh - 6.5vw - 2.5vmax);
    }

    .Administration td, th {
        width: 6.5vw;
        max-width: 6.5vw;
        font-size: 0.8vw;
        height: 1vw;
    }

    .Administration td:last-child {
        width: 2vw;
    }

    .Administration th:last-child {
        width: 3.2vw;
    }

    /*CSS for members array*/
    .Administration .members {
        grid-row: 1 / 3;
        margin: 1vw auto auto;
    }

    /*CSS for CA array*/
    .Administration .CA {
        grid-row: 1 / 3;
        grid-column: 2;
        margin: 1vw auto auto;
    }

    .Administration .memberInfo {
        max-height: calc(85vh - 6.5vw - 2.5vmax);
    }

    .Administration .memberInfo .infoTitle {
        font-size: 1.5vw;
        margin: 1vw 0 1vmin 0;
        text-align: center;
    }

    /*CSS for member info*/
    .Administration .memberInfo .infoArea {
        width: 80%;
        height: 100%;
        border: 0.2vw solid black;
        margin: auto;
    }
}

@media (orientation: portrait) {
    .Administration {
        grid-template-columns: 1fr 1fr;
    }

    /*CSS for members array*/
    .Administration .members {
        grid-row: 1;
        grid-column: 1 / 3;
        margin: 1vw auto auto;
    }

    .Administration caption {
        font-size: 3vw;
    }

    .Administration tbody {
        max-height: 35vh;
    }

    .Administration .members td, .Administration .members th {
        width: 24vw;
        max-width: 24vw;
        font-size: 2vw;
        height: 2.4vw;
    }

    .Administration .members td:last-child {
        width: 6vw;
    }

    .Administration .members th:last-child {
        width: auto;
    }

    /*CSS for CA array*/
    .Administration .CA {
        grid-row: 2;
        grid-column: 1;
        margin: 1vw auto auto;
    }

    .Administration .CA td, .Administration .CA th {
        width: 8.5vw;
        max-width: 8.5vw;
        font-size: 1vw;
        height: 1.2vw;
    }

    .Administration .CA td:last-child {
        width: 3vw;
    }

    .Administration .CA th:last-child {
        width: 3vw;
    }

    .Administration .memberInfo {
        display: none;
    }
}