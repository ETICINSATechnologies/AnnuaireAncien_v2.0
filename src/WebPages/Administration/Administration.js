import React, {Component} from 'react';
import './Administration.css';

import Auth from "../../Components/Auth/Auth";
import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import MemberArray from "../../Components/MemberArray/MemberArray";
import MemberInfo from "../../Components/MemberInfo/MemberInfo";

import {Route, Redirect} from 'react-router'
class Administration extends Component {
    constructor(props) {
        super(props);
        this.state = {
            memberInfo: {}
        };
        this.selectMember = this.selectMember.bind(this)
    }

    selectMember (member) {
        this.setState({memberInfo: member});
    }

    render() {
        if (!Auth.isConnected())
            return <Redirect to='/'/>;

        {/*<Route exact path="/" render={() => (*/}
            {/*if (!Auth.isConnected())*/}
                {/*return <Redirect to='/'/>;*/}
        {/*)}/>*/}

        {/*<Route exact path="/" render={() => (*/}
            {/*loggedIn ? (*/}
            {/*) : (*/}
                {/*<Redirect to="/"/>*/}
            {/*)*/}
        {/*)}/>*/}

        let activeButton = ["home"];
        activeButton.push('search');
        activeButton.push('profile');
        activeButton = Auth.addCorrectButton(activeButton);

        return (
            <React.Fragment>
                <Header/>
                <Nav buttons={activeButton}> </Nav>
                <section className="Administration">
                    <MemberArray className="members" onClick={this.selectMember}/>
                    <MemberArray className="CA" board onClick={this.selectMember}/>
                    <MemberInfo editable info={this.state.memberInfo}/>
                </section>
            </React.Fragment>
        );
    }
}

export default Administration;
