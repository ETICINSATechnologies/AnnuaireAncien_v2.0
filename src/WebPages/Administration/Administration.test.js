import React from 'react';

import Administration from './Administration';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Auth from '../../Components/Auth/Auth';

Enzyme.configure({adapter: new Adapter()});

describe( 'Administration component', () => {
    it('renders without crashing', () => {
            const tree = renderer.create(<Administration />).toJSON();
            expect(tree).toMatchSnapshot();
        }
    );

    it('Administration unit test', () => {
        expect(
            Enzyme.shallow(<Administration/>)
                .find('section.Administration')
                .length
        ).toBe(1)
    });

    it('Administration user not connected test', () => {
            Auth.isConnected = jest.fn(() => false);
            const wrapper = Enzyme.shallow(<Administration />);
            expect( wrapper.find('Redirect').length ).toBe(1);
        }
    );

});