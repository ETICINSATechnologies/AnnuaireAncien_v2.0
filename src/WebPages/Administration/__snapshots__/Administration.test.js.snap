// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Administration component renders without crashing 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/"
      >
        accueil
      </a>
      <a
        className="button"
        href="/search"
      >
        rechercher
      </a>
      <a
        className="button"
        href="/profile"
      >
        mon profil
      </a>
      <a
        className="button"
        href="/connection"
      >
        se connecter
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <section
    className="Administration"
  >
    <section
      className="MemberArray members"
    >
      <table>
        <caption>
          Liste des membres 
          
        </caption>
        <thead>
          <tr
            className="MemberDisplay undefined"
            onClick={[Function]}
          >
            <th
              className="firstName"
            >
              prénom
            </th>
            <th
              className="lastName"
            >
              nom
            </th>
            <th
              className="positionLabel"
            >
              poste
            </th>
            <th
              className="positionYear"
            >
              année
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </section>
    <section
      className="MemberArray CA"
    >
      <table>
        <caption>
          Liste des membres 
          du CA
        </caption>
        <thead>
          <tr
            className="MemberDisplay undefined"
            onClick={[Function]}
          >
            <th
              className="firstName"
            >
              prénom
            </th>
            <th
              className="lastName"
            >
              nom
            </th>
            <th
              className="positionLabel"
            >
              poste
            </th>
            <th
              className="positionYear"
            >
              année
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </section>
    <section
      className="MemberInfo"
    >
      <div
        className="infoTitle"
      >
         Info du membre
      </div>
      
    </section>
  </section>,
]
`;
